src/StateManagers/SizeManager.vala

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2021 Alecaddd (https://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alessandro "Alecaddd" Castellani <castellani.ale@gmail.com>
+ */
+
+ /**
+  * State manager handling the currently selected objects sizings.
+  * This is used to guarantee correct values in the Transform Panel no matter
+  * if one or multiple items are selected.
+  */
+public class Akira.StateManagers.SizeManager : Object {
+    public weak Akira.Window window { get; construct; }
+    private weak Akira.Lib.Canvas canvas;
+
+    // Allow or deny updating the items size.
+    private bool do_update = true;
+
+    // These attributes represent only the primary WIDTH & HEIGHT coordinates of the selected shapes.
+    // These are not the original sizes of each selected shape, but only the BOTTOM-RIGHT values
+    // of the bounding box selection.
+    private double? _width = null;
+    public double width {
+        get {
+            return _width != null ? _width : 0;
+        }
+        set {
+            if (value == _width) {
+                return;
+            }
+
+            _width = Utils.AffineTransform.fix_size (value);
+            update_items_width ();
```

Comment:
I don't really love this get set vala mechanic, I think it's flaky. But thats' just a personal opinion and doesn't matter much. What does matter is to make sure these things never have side effects or events attached to them. I should be able to change values without causing events, etc. Not saying it's teh case here, but want to make sure.

Reply:
That's a good point.
Do you have any suggestion for this situation?
Basically the `width` attribute is binded to the Transform Panel Width field, so every time the user interact with that field, the setter is called and triggers the update of the selected items.
I guess we could bind that field to the width of the select_effect, but that would mean then trickling down the updated width to the items.